sturdycouscous/Garbanzo/Checker.py

```diff
@@ -2,31 +2,50 @@
 # 1) Check certificate validity 
 # 2) Check TLS versions supported
 # 3) Look for open ports
-# 4) Get all ciphers supported (?)
+# 4) Get all ciphers supported 
 import socket, ssl
+import copy
 from datetime import datetime
 
 class connection_checker():
 
-	# Returns a boolean checking whether a specified port for a given domain is open.
-	def port_checker(self, domain, port):
-		s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-		socket.setdefaulttimeout(1)
+	# Returns list of open ports.
+	def port_checker(self, domain):
+
+		port_list = [20, 21, 22, 23, 25, 53, 69, 67, 80, 110, 119, 123, 143, 161, 162, 389, 443, 445, 514, 554, 636, 1720, 3389, 5060, 5061]
+		ports_open = []
+
+		for port in port_list:
+
+			s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+			socket.setdefaulttimeout(1)
 		
-		result = s.connect_ex((domain, port))
-		return (result == 0)
+			result = s.connect_ex((domain, port))
+			if (result == 0):
+				ports_open.append(port)
+			s.close()
+
+		return ports_open
 
 	# Checks whether the certificate obtained using the default_context is still valid.
 	# Returns True if certificate is still valid.
-	def certificate_checker(self, domain):
+	# Tabling the discussion for revoked certificates for later, since it's a bit complicated.
+	def certificate_checker(self, domain, port=443):
 		context = ssl.create_default_context()
 		sslSocket = context.wrap_socket(socket.socket(socket.AF_INET), server_hostname = domain)
-		sslSocket.connect((domain, 443))
 		
-		cert = sslSocket.getpeercert()
-		expiry_date = datetime.strptime(cert['notAfter'], "%b %d %H:%M:%S %Y %Z")
-
-		return datetime.now() < expiry_date
+		# sslSocket.connect can throw an error
+		valid_cert = False
+		try:
+			sslSocket.connect((domain, port))
+			cert = sslSocket.getpeercert()
+			expiry_date = datetime.strptime(cert['notAfter'], "%b %d %H:%M:%S %Y %Z")
+			valid_cert = datetime.now() < expiry_date
+		except:
+			pass
+		finally:
+			sslSocket.close()
+			return valid_cert
 
 
 	# Returns the list of all tls contexts
```

Comment:
Since the previous function returned at line 48, shouldn't there be a new def <function name> here?